onset_maproom/calc.py

```diff
@@ -33,6 +35,80 @@ def onset_date(daily_rain, early_start_day, early_start_month, search_days, rain
 #  onset_date = onset_date[time_coord] + onset_date  
   return onset_date
 
+# Time functions
+
+def daily_groupby_season(daily_data, start_day, start_month, end_day, end_month, time_coord="T"):
+  """Groups daily data by yearly seasons from day-month edges of season
+  Can then apply functions on each group to create yearly time series of seasonal quantities
+  If starting day-month is 29-Feb, uses 1-Mar.
+  If ending day-month is 29-Feb, uses 1-Mar and triggers the option to open the right edge of the Interval.
+  That means that the last day included in the season will be 29-Feb in leap years and 28-Feb otherwise
+  """
+#Deal with leap year cases
+  if start_day == 29 and start_month == 2 :
+    start_day = 1
+    start_month = 3
+  right_bool = True
+  if end_day == 29 and end_month == 2 :
+    end_day = 1
+    end_month = 3
+    right_bool = False
+#Creates array of edges of the season that will form the bins
+  season_bins = daily_data[time_coord].where(
+    ((daily_data[time_coord].dt.day==start_day) & (daily_data[time_coord].dt.month==start_month))
+    |
+    ((daily_data[time_coord].dt.day==end_day) & (daily_data[time_coord].dt.month==end_month)),
+    drop=True
+  )
+#First valid bin can be the 2nd one depending of order of first time_coord point, start/end day-month
+  first_valid_bin = ~((season_bins[0].dt.day == start_day) & (season_bins[0].dt.month == start_month)).values*1
+#First pass to group and then concat only every other season that we want to keep
+  daily_groubedby_season = xr.concat(
+    [daily_data.groupby_bins(time_coord, season_bins, right=right_bool)[g] for g in list(daily_data.groupby_bins(time_coord, season_bins, right=right_bool).groups.keys())[first_valid_bin::2]],
+    time_coord
+  )
+#Second pass to recreate the group labels that concat lost
+  daily_groubedby_season = daily_groubedby_season.groupby_bins(time_coord, season_bins, right=right_bool)
+  return daily_groubedby_season
+
+def seasonal_sum(daily_data, start_day, start_month, end_day, end_month, min_count=None, time_coord="T"):
+  """Calculates seasonal totals of daily data in season defined by day-month edges
+  """
+#It turns out that having daily_groupby_season concatenate only every other group is not enough to drop entired the undesired seasons
+#sum will return NaN for these so need to use dropna to clean up
```

Comment:
Seems like `dropna` might sometimes drop more than you intended. NaNs in the original data might result in legitimate NaNs for the season of interest, and we shouldn't drop those.  If we're going to stick with `groupby_bins` (rather than `where` as I suggested in another comment), we should try to eliminate the unwanted seasons in a more principled way. I won't spend more time thinking about that until you reply about `where`.

Reply:
Correct: dropna could drop more legitimate NAs. In Ingrid, there are clearly 2 distinct worlds between regularly spaced grids and not; but in Python/Xarray, with the labelling system, it was unclear to me. Now to the where question

Reply:
This is another issue that's not resolved, but doesn't need to block us right now. Do you have a todo list somewhere to keep track of issues that we want to carry over after we merge the PR?

Reply:
This is obsolete. We don't need to use dropna anymore because: i) we are not using groupby_bins anymore and ii) we are dropping the daily data out of season before dealing with grouping. Thus if a season is fully missing, it will return NA

Reply:
Is there an issue tracker with gitHub? or how do you keep tracks of to-do list on your other coding projects?

Reply:
Oh right, we don't use dropna anymore. This is resolved.